### DISCUSSION TITLE
Add user to database using keycloak's user registration flow.

### DISCUSSION THREAD

**Discussion by BhaskarAkash** [keycloak/keycloak#11129] [Q&A]
I am trying to add a new user to database using keycloak's user registration with a custom provider. I implemented UserRegistrationProvider interface and have overrode the addUser() method which only takes Username as argument to save this user to database. However i want to store all the user details (firstname, lastname, password, email) to database. Could anybody suggest any implementation which can serve the purpose?

**Comment by javac9** [comment]
Hi, I'm having the same problem, just asked here: https://github.com/keycloak/keycloak/discussions/16073
Let's hope someone provides some input!

**Reply by javac9** [reply]
@BhaskarAkash Just stumbled upon: https://www.tabnine.com/code/java/methods/org.keycloak.models.UserProvider/addUser
It might help you.
